Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.cpp

```diff
@@ -143,51 +203,447 @@ KeyboardMouse::KeyboardMouse(const LPDIRECTINPUTDEVICE8 kb_device,
-  m_state_in.cursor.x = (ControlState(point.x) / win_width * 2 - 1) * window_scale.x;
-  m_state_in.cursor.y = (ControlState(point.y) / win_height * 2 - 1) * window_scale.y;
+      break;
+    case WEST:
+      return NORTH_WEST;
+      break;
+    case SOUTH_WEST:
+
+      break;
+    }
+  }
+  if (mouse_point.x < octagon_points[index_of_min_octagon_point].x)
+  {
+    switch (index_of_min_octagon_point)
+    {
+    case SOUTH:
+      return SOUTH_WEST;
+      break;
+    case SOUTH_EAST:
+      return SOUTH;
+      break;
+    case EAST:
+
+      break;
+    case NORTH_EAST:
+      return NORTH;
+      break;
+    case NORTH:
+      return NORTH_WEST;
+      break;
+    case NORTH_WEST:
+      return WEST;
+      break;
+    case WEST:
+
+      break;
+    case SOUTH_WEST:
+      return WEST;
+      break;
+    }
+  }
+  if (mouse_point.y > octagon_points[index_of_min_octagon_point].y)
+  {
+    switch (index_of_min_octagon_point)
+    {
+    case SOUTH:
+
+      break;
+    case SOUTH_EAST:
+
+      break;
+    case EAST:
+      return SOUTH_EAST;
+      break;
+    case NORTH_EAST:
+
+      break;
+    case NORTH:
+
+      break;
+    case NORTH_WEST:
+
+      break;
+    case WEST:
+      return SOUTH_WEST;
+      break;
+    case SOUTH_WEST:
+
+      break;
+    }
+  }
+  if (mouse_point.x > octagon_points[index_of_min_octagon_point].x)
+  {
+    switch (index_of_min_octagon_point)
+    {
+    case SOUTH:
+      return SOUTH_EAST;
+      break;
+    case SOUTH_EAST:
+      return EAST;
+      break;
+    case EAST:
+
+      break;
+    case NORTH_EAST:
+      return EAST;
+      break;
+    case NORTH:
+      return NORTH_EAST;
+      break;
+    case NORTH_WEST:
+      return NORTH;
+      break;
+    case WEST:
+
+      break;
+    case SOUTH_WEST:
+      return SOUTH;
+      break;
+    }
+  }
+  return 0;
+}
+
+void KeyboardMouse::Move_Mouse_Point_Along_Gate(POINT& mouse_point, const POINT octagon_points[8])
+{
+  // Sage 4/2/2022: Gameplan is to move the mouse towards the nearest gate if the distance the mouse
+  // is outside of the octagon is large enough to overcome friction.The amount the mouse will move
+  // towards the gate will be the distance outside of the octagon minus friction. This is to
+  // simulate the force required to get a real stick into gates while keeping the ability to hit the
+  // values in between gates consistently. If the distance outside of the octagon is not large
+  // enough to overcome friction, the mouse will get moved to the closest point inside of the
+  // octagon. This is almost always going to be directly on the line between octagon points. I may
+  // need to bring the mouse a tiny bit inside of the octagon instead of directly on the nearest
+  // line if the section of the code that determines whether or not the mouse is inside of the
+  // octagon struggles with the mouse_point being directly on the line.
+  long distance_to_octagon_points[8] = {0};
```

Comment:
```suggestion
  std::array<long, 8> long distance_to_octagon_points{};
```

this allows the size of the array to be directly queried with `.size()`, so the size of the array doesn't need to be hardcoded into the loops below